### PR TITLE
Feature/xaptc 41  --POST Test Case

### PR DESCRIPTION
**Pull request by frankquinn**

### CODE REVIEW

**Comment by frankquinn** [comment]
**** This needs to be merged before XAPTC-44 (GET a Test Case) can be merged *****

File: uiServices/src/testCaseServices_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+   "responseStatusCode":200,
+   "responseContentType": "JSON" ,
+   "preThinkTime": 1000,
+   "postThinkTime":2000,
+   "execWeight": "Sparse"
+}
+`
+
+
+func TestValidTestCasePost(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (frankquinn) [reply uiServices/src/testCaseServices_test.go]: GoLang will handle this

File: ui-src/src/assets/testCase_schema.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
+{
+  "type": "object",
```
Comment (jenny-flynn) [inline ui-src/src/assets/testCase_schema.json]: add ```"additionalProperties": false,``` at the top level too

File: uiServices/src/testCaseServices.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"github.com/Sirupsen/logrus"
+	"github.com/xtracdev/automated-perf-test/testStrategies"
+	"bytes"
+)
+
+var testCaseSchema string = "testCase_schema.json"
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices.go]: these should be ```const``` not ```var```

File: uiServices/src/testCaseServices_test.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, http.StatusCreated, w.Code, "Error: Did Not Successfully Post")
+}
+
+func TestCasePostWithExitsingFileName(t *testing.T) {
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices_test.go]: Existing

File: uiServices/src/testCaseServices_test.go [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	r.HandleFunc("/test-cases", postTestCase)
+
+	filePath := ""
+	request, err := http.NewRequest(http.MethodPost, "/test-cases", reader)
+	request.Header.Set("testCasePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
```
Comment (jenny-flynn) [inline uiServices/src/testCaseServices_test.go]: replace all of these with assert.NotNil or assert.NoError

File: uiServices/src/xmlWriter.go [hunk 6]
```diff
@@ -53,3 +53,24 @@ func testSuiteWriterXml(testSuite testStrategies.TestSuite, configPathDir string
...
+	file, err := os.Create(filename)
+	if err != nil {
+		log.Error("Failed to create output file. Error:", err)
+		return false
+	}
+	if file != nil {
+		defer file.Close()
+		file.Write(testCaseAsXml)
+	}
+	return true
```
Comment (jenny-flynn) [inline uiServices/src/xmlWriter.go]: if file is nil, you're returning true.

switch it around
```
if file == nil {
     log.Error("File was not created successfully - was nil")
     return false
}
defer file.Close()
file.Write(testCaseAsXml)
return true
```

and you should check that file.Write doesn't return an error either.

File: uiServices/test/testCase.feature [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+   	 "Key": "Authorization",
+     "Value" :"Header-Value"
+   }],
+  "ResponseValues":[{
+     "Value":"Res-Value",
+     "ExtractionKey": "Res-Key"
+  }],
+  "MultipartPayload":[{
+     "fieldName": "F-Name",
+   	 "FieldValue":"PayloadName",
```
Comment (jenny-flynn) [inline uiServices/test/testCase.feature]: formatting

**Comment by frankquinn** [comment]
Changes implemented

**Comment by frankquinn** [comment]
All changes implemented, all conflicts fixed. Ready for merging.   @ConorSmyth

**Pull request merged**